CRM/ManageLetterheads/Form/LetterheadForm.php

```diff
@@ -0,0 +1,211 @@
+      return;
+    }
+
+    $letterhead = civicrm_api3('Letterhead', 'getsingle', [
+      'id' => $this->_letterheadId,
+    ]);
+
+    $letterhead['available_for'] = $this->getAvailableForValuesAsKeys(
+      $letterhead['available_for']
+    );
+
+    $this->defaultValues = $letterhead;
+
+    return $this->defaultValues;
+  }
+
+  /**
+   * Handles the letterhead creation and update process.
+   */
+  public function postProcess() {
+    $formValues = $this->controller->exportValues($this->_name);
+    $isUpdateAction = $this->_action & CRM_Core_Action::UPDATE;
+
+    if ($isUpdateAction) {
+      $formValues['id'] = $this->_letterheadId;
+    }
+
+    $this->createLetterhead($formValues);
+    $this->addSuccessMessage();
+    $this->redirectToLetterheadsListPage();
+  }
+
+  /**
+   * Flips an array of Available For values as keys of the array.
+   *
+   * This is done because they are stored as [a, b, c], where a, b, and are
+   * the Available For option values, but the form understands it as
+   * [a => 1, b => 0, c => 1] where 1 or 0 means checked/uncheked.
+   *
+   * @param array $availableFor
+   *   An array Available For option values.
+   * @return array
+   */
+  private function getAvailableForValuesAsKeys(array $availableFor) {
+    $availableFor = array_flip($availableFor);
+
+    return array_map(function () {
+      return '1';
+    }, $availableFor);
+  }
+
+  /**
+   * Creates the letterhead using the given form values.
+   */
+  private function createLetterhead($formValues) {
+    $letterhead = [
+      'title' => $formValues['title'],
+      'description' => $formValues['description'],
+      'content' => $formValues['content'],
+      'available_for' => array_keys($formValues['available_for']),
+      'weight' => $formValues['weight'],
+      'is_active' => isset($formValues['is_active']) ? '1' : '0',
+    ];
+
+    civicrm_api3('Letterhead', 'create', $letterhead);
+  }
+
+  /**
+   * Displays a success message after the letterhead has been created.
+   */
+  private function addSuccessMessage() {
+    CRM_Core_Session::setStatus(
+      ts('Letterhead has been saved.'),
+      ts('Saved'),
+      'success'
+    );
+  }
+
+  /**
+   * Redirects the user to the list page.
+   */
+  private function redirectToLetterheadsListPage() {
+    $url = CRM_Utils_System::url('civicrm/letterheads/list', NULL, TRUE);
+
+    CRM_Utils_System::redirect($url);
+  }
+
+  /**
+   * Adds the form elements including fields and buttons.
+   */
+  private function addFormElements() {
+    $this->add('text', 'title', ts('Title'), NULL, TRUE);
+    $this->add('textarea', 'description', ts('Description'));
+    $this->add('wysiwyg', 'content', ts('Content'), NULL, TRUE);
+    $this->addCheckbox(
+      'available_for',
+      ts('Available For'),
+      array_flip(LetterheadAvailability::buildOptions('available_for')),
+      NULL,
+      NULL,
+      TRUE
+    );
+    $this->add('text', 'weight', ts('Order'));
```

Comment:
We can use this field here: 
`$this->add('number', 'weight', ts('Order'), [], TRUE);`
This will make sure that only numbers are inputted and also add an arrow for increasing/decreasing the weight field.